### PR TITLE
[Command] Add shortcut methods to access container services

### PR DESCRIPTION
**Pull request by jdecool**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

This PR aim to add shortcuts to access container services like it could be done in controllers. I think it could be improve the DX.

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php [hunk 1]
```diff
@@ -53,4 +53,40 @@ public function setContainer(ContainerInterface $container = null)
...
     }
+
+    /**
+     * Returns true if the service id is defined.
+     *
+     * @param string $id The service id
+     *
+     * @return bool true if the service id is defined, false otherwise
+     */
+    public function has($id)
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: it should not be public but protected

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php [hunk 2]
```diff
@@ -53,4 +53,40 @@ public function setContainer(ContainerInterface $container = null)
...
+    {
+        return $this->container->has($id);
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: $this->container might not be initialized. so it needs to be $this->getContainer

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php [hunk 3]
```diff
@@ -53,4 +53,40 @@ public function setContainer(ContainerInterface $container = null)
...
+    }
+
+    /**
+     * Gets a container service by its id.
+     *
+     * @param string $id The service id
+     *
+     * @return object The service
+     */
+    public function get($id)
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: exceptions that can be thrown are missing in the phpdoc

**Comment by Tobion** [comment]
:-1: It does not add any actual logic but just magic through indirection.

**Comment by jdecool** [comment]
I agree with that PR doesn't had any logic. I only syntaxic sugar for developer to simplify code when using the DIC.

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php [hunk 4]
```diff
@@ -53,4 +53,46 @@ public function setContainer(ContainerInterface $container = null)
...
     }
+
+    /**
+     * Returns true if the service id is defined.
+     *
+     * @param string $id The service id
+     *
+     * @return bool true if the service id is defined, false otherwise
+     *
+     * @throws \LogicException
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: these exceptions are wrong

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php [hunk 5]
```diff
@@ -53,4 +53,46 @@ public function setContainer(ContainerInterface $container = null)
...
+    }
+
+    /**
+     * Gets a container service by its id.
+     *
+     * @param string $id The service id
+     *
+     * @return object The service
+     *
+     * @throws \LogicException
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: too generic and some are missing

File: src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php (hunk 4 above)
Comment (jdecool) [inline src/Symfony/Bundle/FrameworkBundle/Command/ContainerAwareCommand.php]: What is wrong with this one ?

This exception is actually return by the `getContainer` method.

**Comment by fabpot** [comment]
I'm also :-1:

**Pull request closed without merging**